spec/overcommit/hook_context/pre_push_spec.rb

```diff
@@ -44,6 +44,75 @@
     end
   end
 
+  describe '#modified_files' do
+    let(:remote_url) { "file://#{File.expand_path('some-repo')}" }
+
+    subject { context.modified_files }
+
+    context 'when current branch has tracking branch' do
+      let(:local_ref) { 'refs/heads/project-branch' }
+      let(:local_sha1) { get_sha1(local_ref) }
+      let(:remote_ref) { 'refs/remotes/origin/master' }
+      let(:remote_sha1) { get_sha1(remote_ref) }
+      let(:input) do
+        double('input', read: "#{local_ref} #{local_sha1} #{remote_ref} #{remote_sha1}\n")
+      end
+
+      it 'has modified files based on tracking branch' do
+        with_remote_repo do
+          `git checkout -b project-branch 2>&1 > #{File::NULL}`
+          `git push -u origin project-branch 2>&1 > #{File::NULL}`
+
+          touch 'added-1'
+          `echo add >> added-1`
+          `echo append >> update-me`
+          `rm delete-me`
```

Comment:
`FileUtils.rm` (again to avoid a child process).

Reply:
👍🏻